history/src/block_stores/inmemory_block_store.rs

```diff
@@ -0,0 +1,71 @@
+use async_trait::async_trait;
+use solana_lite_rpc_core::{
+    structures::produced_block::ProducedBlock,
+    traits::block_storage_interface::BlockStorageInterface,
+};
+use solana_rpc_client_api::config::RpcBlockConfig;
+use solana_sdk::slot_history::Slot;
+use std::{collections::BTreeMap, ops::Range};
+use tokio::sync::RwLock;
+
+pub struct InmemoryBlockStore {
+    block_storage: RwLock<BTreeMap<Slot, ProducedBlock>>,
+    number_of_blocks_to_store: usize,
+}
+
+impl InmemoryBlockStore {
+    pub fn new(number_of_blocks_to_store: usize) -> Self {
+        Self {
+            number_of_blocks_to_store,
+            block_storage: RwLock::new(BTreeMap::new()),
+        }
+    }
+
+    pub async fn store(&self, slot: Slot, block: ProducedBlock) -> Option<ProducedBlock> {
+        let mut block_storage = self.block_storage.write().await;
+        let min_slot = match block_storage.first_key_value() {
+            Some((slot, _)) => *slot,
+            None => 0,
+        };
+        if slot > min_slot {
+            block_storage.insert(slot, block);
+            if block_storage.len() > self.number_of_blocks_to_store {
+                block_storage.remove(&min_slot)
+            } else {
+                None
```

Comment:
looks wrong! shouldn't the method return the stored block?

Reply:
this function was suppose to return the deleted block.